### PR TITLE
Fix unit tests that are actually integration tests

### PR DESCRIPTION
**Pull request by deejgregor**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14462

### CODE REVIEW

**Comment by deejgregor** [comment]
I should add that this gets `./compile.pl` without `-DskipTests` much closer to working. There are some other issues unrelated to tests to be worked out (those will be coming along in separate PRs).

File: core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java [hunk 1]
```diff
@@ -366,8 +370,10 @@ public void testTimeoutNoRetries() throws Exception {
 
         long elapsedTime = cb.timeoutTimestamp - req.getSentTimestamp();
 
+        System.out.println("testTimeoutNoRetries processing took " + elapsedTime);
```
Comment (chuklov) [inline core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java:373]: I would change here to use LOG.info() instead of any System.out.print

**Review by chuklov — CHANGES_REQUESTED** [review]

File: core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java [hunk 2]
```diff
@@ -400,10 +406,10 @@ public void testTimeoutOneRetry() throws Exception {
 
         long elapsedTime = cb.timeoutTimestamp - req.getSentTimestamp();
 
-        // no more than two millis should pass before the timeout is processed
-        assertTrue( "Timeout processed too late", elapsedTime < (2 * TIMEOUT + 10) );
-
+        System.out.println("testTimeoutOneRetry processing took " + elapsedTime);
```
Comment (chuklov) [inline core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java:409]: Here as well

**Review by chuklov — CHANGES_REQUESTED** [review]

File: core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java (hunk 1 above)
Comment (deejgregor) [reply core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java]: Updated and force pushed.

File: core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java (hunk 2 above)
Comment (deejgregor) [reply core/lib/src/test/java/org/opennms/core/tracker/RequestTrackerTest.java]: Updated and force pushed.

**Review by chuklov — APPROVED** [review]

**Pull request merged**